### ISSUE TITLE
Respect SublimeLinter settings

### ISSUE THREAD

**Issue by goodguyry** [Rockstar04/SublimeLinter-contrib-phpstan#21]
While trying to understand why this package doesn't appear to do anything, I noticed my `--configuration` and `--autoload-file` settings are duplicated in the command run by the linter.

Project settings:
```json
"SublimeLinter.linters.phpstan.executable": "${folder}/client-mu-plugins/vendor/bin/phpstan",
"SublimeLinter.linters.phpstan.args": [
  "--memory-limit=1G",
  "--configuration ${folder}/phpstan.neon",
  "--autoload-file=${folder}/client-mu-plugins/vendor/autoload.php"
],
```

Debugging console output (formatted for clarity & with paths replaced by ST variables):
```
${folder}/client-mu-plugins/vendor/bin/phpstan analyse \
  --memory-limit=1G \
  --configuration ${folder}/phpstan.neon \
  --autoload-file=${folder}/client-mu-plugins/vendor/autoload.php. \
  --error-format=json \
  --no-progress \
  --configuration=${project_path}/phpstan.neon \
  --autoload-file=${project_path}/vendor/autoload.php -- <path-to-file>
```

Because I'm working in `${project_path}`, this package assumes my `.neon` and `autoload.php` files should be in that directory in spite of my settings explicitly stating otherwise.

Relevant code: https://github.com/Rockstar04/SublimeLinter-contrib-phpstan/blob/master/linter.py#L26-L40

**To summarize:** This package is including the SublimeLinter config `args` from my project settings, and it is appending some "helpful" defaults, which results in the command not working. If I run the command in Terminal, I see PHP fatal errors; if I run the same command with the duplicated args removed, the command works as expected.

As-is, this plugin doesn't appear to respect SublimeLinter settings. It should check `args` for those options before attempting to patch them in.

**Comment by pierresh** [comment]
Hello, 

yes, this plugin does not use SublimeLinter config because finding the `phpstan.neon` file makes sure that it uses the related project settings.

Example, if you open a file from the project A, it will lint according to project's A settings, and if you open a file from the project B, it will lint according to project's A settings.

**Comment by goodguyry** [comment]
The purpose of the config is to allow users to specify how their project structure differs from the norm/expected. I can fully understand attempting to find those files if there's nothing in the config pointing to them. However, as this is a package for use with SublimeLinter, it should respect users' SublimeLinter config.

**Comment by kaste** [comment]
That is not so complicated to support.  a) Don't add special settings, let the user put whatever they want in (the by default available, catch-all) `args` argument, and check in the code (`cmd`) what is set and what is missing.   b)  *If* `--configuration` and `--autoload-file` are more or less mandatory, define explicit settings (setting names) for them:



```
    defaults = {
        "--configuration": "",

    ...

    def cmd():
        ...
        if self.settings.get('configuration') == '':
            ... # find the configuration
            self.settings.set('configuration', <path>)
        ...
```

The latter allows users to set `configuration` in their settings.  Or to set it
to `false` which will skip the argument without triggering the automatic search.

The downside is that it is technically a breaking change.

**Comment by pierresh** [comment]
I think I found a simple solution for this use case, [I just committed it.](https://github.com/Rockstar04/SublimeLinter-contrib-phpstan/commit/a537b5e08ad027ee6b53794fe94253b9dae1a218)

@goodguyry Please try it and let me know if it works for you, thanks

**Comment by goodguyry** [comment]
Sure, I can find time this week to test it out. Thanks

**Comment by goodguyry** [comment]
I do see my config args in the command with `debug` enabled. Sadly, I introduced an intentional error and the plugin does not show any messaging in the editor. That appears to be unrelated, since I can run the command from the ST console in Terminal and see the JSON output, so I'd consider this issue fixed.
